In [162]:
from google.colab import drive
drive.mount('/content/drive')
%cd /content/drive/MyDrive/

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.activity.readonly&response_type=code

Enter your authorization code:
q


ValueError: ignored

In [227]:
# Imports

import torch
import torch.nn as nn
import pandas as pd
import numpy as np
import re
from IPython.display import HTML
from sklearn.feature_extraction.text import CountVectorizer
from torch.utils.data import Dataset, random_split
from sklearn.feature_extraction.text import TfidfTransformer
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
import codecs

In [228]:
# Setting random seed and device
SEED = 1

torch.manual_seed(SEED)
torch.cuda.manual_seed(SEED)
torch.backends.cudnn.deterministic = True

use_cuda = torch.cuda.is_available()
device = torch.device("cuda:0" if use_cuda else "cpu")

In [229]:
# Full width display
pd.set_option('display.max_colwidth', None)
pd.set_option('display.width', None)

# TASK 1

In [230]:
# Load data
train_df = pd.read_csv('train.csv', dtype={'grades': 'string'})
train_df['grades'] = train_df['grades'].str.replace('', ' ').str.strip()

#test_df = pd.read_csv('data/task-1/dev.csv')
#test_df['grades'] = test_df['grades'].str.replace('', ' ').str.strip()

In [231]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9652 entries, 0 to 9651
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   id         9652 non-null   int64  
 1   original   9652 non-null   object 
 2   edit       9652 non-null   object 
 3   grades     9652 non-null   string 
 4   meanGrade  9652 non-null   float64
dtypes: float64(1), int64(1), object(2), string(1)
memory usage: 377.2+ KB


In [232]:
#test_df.info()

In [233]:
print(train_df['original'][0])
print()
train_df.head()

France is ‘ hunting down its citizens who joined <Isis/> ’ without trial in Iraq



id  ... meanGrade
0  14530  ...       0.2
1  13034  ...       1.6
2   8731  ...       1.0
3     76  ...       0.4
4   6164  ...       0.0

[5 rows x 5 columns]

In [236]:
# Expanding the stupidly jointed grades

grade_names = []
for i in range(15):
    grade_names.append(f'grade {i+1}')
train_df[grade_names] = train_df['grades'].str.split(' ', expand=True)

# Turn valid grades to int, missing ones to NaN
train_df[grade_names] = train_df[grade_names].apply(pd.to_numeric, errors='coerce')

In [237]:
train_df.head()

# Whole df as html 
# HTML(train_df.to_html())

id  ... grade 15
0  14530  ...      NaN
1  13034  ...      NaN
2   8731  ...      NaN
3     76  ...      NaN
4   6164  ...      NaN

[5 rows x 20 columns]

In [239]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9652 entries, 0 to 9651
Data columns (total 20 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   id         9652 non-null   int64  
 1   original   9652 non-null   object 
 2   edit       9652 non-null   object 
 3   grades     9652 non-null   string 
 4   meanGrade  9652 non-null   float64
 5   grade 1    9652 non-null   int64  
 6   grade 2    9652 non-null   int64  
 7   grade 3    9652 non-null   int64  
 8   grade 4    9652 non-null   int64  
 9   grade 5    9652 non-null   int64  
 10  grade 6    142 non-null    float64
 11  grade 7    142 non-null    float64
 12  grade 8    142 non-null    float64
 13  grade 9    142 non-null    float64
 14  grade 10   142 non-null    float64
 15  grade 11   6 non-null      float64
 16  grade 12   6 non-null      float64
 17  grade 13   6 non-null      float64
 18  grade 14   6 non-null      float64
 19  grade 15   6 non-null      float64
dtypes: float

How to best handle the 6 with 15 grads and the 142 with 10 grades?

In [190]:
train_df.sort_values('meanGrade',ascending=False)[['original','edit', 'meanGrade']]

original  ... meanGrade
6075                                           Recent Scandals Highlight Trump 's Chaotic <Management/> Style  ...       3.0
1490                     President Trump 's first year <anniversary/> report card , with grades from A + to F  ...       3.0
4438                           How an FBI raid fed a rumor that Orrin Hatch was about to become <president/>   ...       2.8
8726                                                 Trump admits tariffs could cause ' pain ' in <markets/>   ...       2.8
6946                                                                Kasich : Trump <tweets/> ‘ unacceptable ’  ...       2.8
...                                                                                                       ...  ...       ...
9250  It did n’t end at the ballot box : <Donald Trump/> ’s biggest supporters now push for a divisive agenda  ...       0.0
1237              Delhi Police Say They 've Captured Most-Wanted Terrorist Known As ' <India/> 's Bin Laden '  ...       0.0
9253                     Trump 's Climate-Denying Coal Lobbyist Nominee Inches Closer To EPA ’s No. 2 <Job/>   ...       0.0
4421                                                     Democratic division simmers at feel-good <retreat/>   ...       0.0
2885                                   U.S. Politics Are Rigged . It 's Time For a Small-Donor <Revolution/>   ...       0.0

[9652 rows x 3 columns]

# TASK 2

In [ ]:
# Load data
train_df = pd.read_csv('data/task-2/train.csv')
test_df = pd.read_csv('data/task-2/dev.csv')

In [ ]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9381 entries, 0 to 9380
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id          9381 non-null   object 
 1   original1   9381 non-null   object 
 2   edit1       9381 non-null   object 
 3   grades1     9381 non-null   int64  
 4   meanGrade1  9381 non-null   float64
 5   original2   9381 non-null   object 
 6   edit2       9381 non-null   object 
 7   grades2     9381 non-null   int64  
 8   meanGrade2  9381 non-null   float64
 9   label       9381 non-null   int64  
dtypes: float64(2), int64(3), object(5)
memory usage: 733.0+ KB


In [ ]:
train_df

id                                          original1  \
0      10920-9866  " Gene Cernan , Last <Astronaut/> on the Moon ...   
1      3176-10722  " I 'm done " : Fed up with California , some ...   
2       3176-3702  " I 'm done " : Fed up with California , some ...   
3      10722-3702  " I 'm done " : Fed up with <California/> , so...   
4      12282-2083  " Our expectations of what civic engagement lo...   
...           ...                                                ...   
9376    975-13357  “ It ’s painfully obvious " Mueller will charg...   
9377    975-11773  “ It ’s painfully obvious " Mueller will charg...   
9378  13357-11773  “ It ’s painfully obvious " Mueller will charg...   
9379  14954-14479  “ Kompromat , ” media ethics and the law : Wha...   
9380   6354-12970  “ Son of a bitch ” : Trump ’s NFL <outburst/> ...   

         edit1     grades1  meanGrade1  \
0       Dancer        1113         1.2   
1     vagrants        1200         0.6   
2     vagrants        1200         0.6   
3     pancakes       10110         0.6   
4      imagine           0         0.0   
...        ...         ...         ...   
9376   battery           1         0.2   
9377   battery           1         0.2   
9378  plumbing       11103         1.2   
9379      porn       20101         0.8   
9380   tantrum  3111130000         1.0   

                                              original2        edit2  \
0     " Gene Cernan , Last Astronaut on the Moon , <...  impregnated   
1     " I 'm done " : Fed up with <California/> , so...     pancakes   
2     " I 'm done " : Fed up with <California/> , so...         life   
3     " I 'm done " : Fed up with <California/> , so...         life   
4     " Our expectations of what civic engagement <l...       smells   
...                                                 ...          ...   
9376  “ It ’s painfully obvious " Mueller will charg...     plumbing   
9377  “ It ’s painfully obvious " Mueller will <char...     strangle   
9378  “ It ’s painfully obvious " Mueller will <char...     strangle   
9379  “ Kompromat , ” media ethics and the law : Wha...        dance   
9380  “ Son of a <bitch/> ” : Trump ’s NFL outburst ...       father   

        grades2  meanGrade2  label  
0         30001         0.8      1  
1         10110         0.6      0  
2             2         0.4      1  
3             2         0.4      1  
4     100220010         0.6      2  
...         ...         ...    ...  
9376      11103         1.2      2  
9377      22331         2.2      2  
9378      22331         2.2      2  
9379      32112         1.8      2  
9380      10000         0.2      1  

[9381 rows x 10 columns]

In [ ]:
funniest = train_df.sort_values('meanGrade1',ascending=False)[['original1','edit1','meanGrade1','edit2']]

In [ ]:
for i in range(5):
    print(funniest['original1'][i],funniest['edit1'][i],' vs ',funniest['edit2'][i])

" Gene Cernan , Last <Astronaut/> on the Moon , Dies at 82 " Dancer  vs  impregnated
" I 'm done " : Fed up with California , some <conservatives/> look to Texas vagrants  vs  pancakes
" I 'm done " : Fed up with California , some <conservatives/> look to Texas vagrants  vs  life
" I 'm done " : Fed up with <California/> , some conservatives look to Texas pancakes  vs  life
" Our expectations of what civic engagement looks like do n’t match reality . Can we <fix/> that ? " imagine  vs  smells
